src/runtime/composables/useSupabaseUser.ts

```diff
@@ -1,9 +1,28 @@
 import type { User } from '@supabase/supabase-js'
 import { useSupabaseSession } from './useSupabaseSession'
-import { computed, type ComputedRef } from '#imports'
+import { useSupabaseClient } from './useSupabaseClient'
+import { useState } from '#imports'
+
+export const useSupabaseUser = async () => {
+  const userState = useState<User | null>('supabase_user', () => null)
+  const supabase = useSupabaseClient()
+
+  // Need to manipulate user in `supabase.client` plugin when client is not yet initialized
+  if (!supabase) {
+    return userState
+  }
+
+  // We do not rely on `getSession` on the server. It could be tampered with by the sender.
+  if (import.meta.server) {
+    const { data: { user } } = await supabase.auth.getUser()
+    userState.value = user
+  }
```

Comment:
I recommend to move it inside a server plugin to avoid having an async `useSupabaseUser()` and avoid fetching the API on each time we call the `useSupabaseUser()`

Reply:
@larbish are you going to tackle this (since you made the async change in the first place) or should I take a stab at it?

Reply:
I'll do it, we discussed it together with @Atinux

Reply:
Done in https://github.com/nuxt-modules/supabase/pull/357/commits/f7cbc2fdeaf4a32e164c843be2a312db925e916d

As discussed together @Atinux, I have to keep `useSupabaseUser()` async since it's using the async `useSupabaseSession`.

@felixgabler Can u ensure everything still works as expected on your project?